pipeline/authn/authenticator_bearer_token.go

```diff
@@ -71,6 +86,14 @@ func (a *AuthenticatorBearerToken) Config(config json.RawMessage) (*Authenticato
 	if len(c.SubjectFrom) == 0 {
 		c.SubjectFrom = "sub"
 	}
+	if len(c.ForwardHTTPHeaders) == 0 {
+		c.ForwardHTTPHeaders = []string{header.Authorization, header.Cookie}
+	}
```

Comment:
Just to make sure I'm understanding this correctly, if we have no `forward_http_headers` we would always add `Authorization` and `Cookie`. But shouldn't that be the case for `bearer` and `cookie` authenticators anyway, even if we add more `forward_http_headers`?

referring to this discussion https://github.com/ory/oathkeeper/pull/952#discussion_r868183613

Reply:
Yeah. I found the issue here. Right now oathkeeper adds `Authorization` and `Cookie` headers only when we have no `forward_http_headers` configured. However, I think that the oathkeeper should have them in any case because they're essential. 

The issue may occur when someone configures something like
```
forward_http_headers:
   - Accept
   - X-User
```
In that case, the authentication flow breaks because the oathkeeper does not proxies essential headers.

Reply:
I think that we can remove `if len(c.ForwardHTTPHeaders) == 0` and append `[]string{header.Authorization, header.Cookie}` anyway. WDYT?

Reply:
i agree since it is required for `bearer` or `cookie` authenticator to work properly

Reply:
Fixed it too